pkg/controllers/sync/controller.go

```diff
@@ -945,7 +935,7 @@ func (s *SyncController) handleDeletionInClusters(
 		return false, errors.Wrap(err, "failed to get a list of clusters")
 	}
 
-	dispatcher := dispatch.NewUnmanagedDispatcher(s.informer.GetClientForCluster, gvk, qualifiedName)
+	dispatcher := dispatch.NewUnmanagedDispatcher(s.logger, s.informer.GetClientForCluster, gvk, qualifiedName)
```

Comment:
`NewCheckUnmanagedDispatcher` takes in a ctx while `NewUnmanagedDispatcher` takes in a logger. Why the inconsistency?

Reply:
The context currently used by all dispatchers, please take a review.